File: server/src/main/java/ua/com/alevel/service/finance/AttendanceService.java
package ua.com.alevel.service.finance;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.finance.Attendance;
import ua.com.alevel.persistence.repository.finance.AttendanceRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class AttendanceService extends AbstractService<Attendance, AttendanceRepository> {

    private final AttendanceRepository attendanceRepository;
    private final CrudRepositoryHelper<Attendance, AttendanceRepository> crudRepositoryHelper;

    public AttendanceService(AttendanceRepository attendanceRepository, CrudRepositoryHelper<Attendance, AttendanceRepository> crudRepositoryHelper) {
        super(attendanceRepository, crudRepositoryHelper);
        this.attendanceRepository = attendanceRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}

File: client-admin/src/app/shared/services/finance/attendance.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Attendance} from "../../models/finance/attendance.model";

@Injectable({
  providedIn: 'root'
})
export class AttendanceService extends AbstractService<Attendance> {

  override resourceUrl = '/api/attendances';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}
